src/backend/query_builder.rs

```diff
@@ -58,21 +58,27 @@ pub trait QueryBuilder: QuotedBuilder {
         sql: &mut SqlWriter,
         collector: &mut dyn FnMut(Value),
     ) {
-        write!(sql, "SELECT ").unwrap();
+        let is_exists_stmt = select.exists.is_some();
 
-        if let Some(distinct) = &select.distinct {
-            write!(sql, " ").unwrap();
-            self.prepare_select_distinct(distinct, sql, collector);
-            write!(sql, " ").unwrap();
-        }
+        write!(sql, "SELECT ").unwrap();
 
-        select.selects.iter().fold(true, |first, expr| {
-            if !first {
-                write!(sql, ", ").unwrap()
+        if is_exists_stmt {
+            write!(sql, "EXISTS (SELECT * ").unwrap();
```

Comment:
First of all thank you! I liked the API overall. However, seeing the code generation part here makes me feel like something is missing. We definitely need to fully express this concept in the syntax tree.
We could probably expand the definition of `SimpleExpr::SubQuery`

```rust
SelectStatement {
    selects: [
        SelectExpr {
            expr: SimpleExpr::SubQuery(
                SubQueryOper::Exists,
                SelectStatement{ ... }
            ),
            alias: None
        }
    ]
    ...
}
```